databases/week2/homework-part4.sql

```diff
@@ -0,0 +1,40 @@
+# COFFEE SHOP DATABASE
+
+SET NAMES utf8mb4;
+
+CREATE DATABASE coffe_shop;
+USE coffe_shop;
+
+CREATE TABLE `product` (
+  `id` int(10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  `name` varchar(255) NOT NULL,
+  `details` text,
+  `price` int(10) unsigned NOT NULL
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+CREATE TABLE `order` (
+  `id` int(10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  `customer_id` int(10) unsigned NOT NULL,
+  `payment_method` varchar(255) NOT NULL
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+CREATE TABLE `order_product` (
+  `order_id` int(10) unsigned NOT NULL,
+  `product_id` int(10) unsigned NOT NULL,
+  PRIMARY KEY(`order_id`, `product_id`),
+  CONSTRAINT `fk_order_product_order` FOREIGN KEY (`order_id`) REFERENCES `order` (`id`) ON DELETE CASCADE ON UPDATE CASCADE,
+  CONSTRAINT `fk_order_product_product` FOREIGN KEY (`product_id`) REFERENCES `product` (`id`) ON DELETE CASCADE ON UPDATE CASCADE
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+CREATE TABLE `status` (
+  `id` int(10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  `status` varchar(255) NOT NULL
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+CREATE TABLE `supplier` (
```

Comment:
Are you sure the supplier can only take care of one order?

Reply:
the supplier can take multiple orders, but 1 order cannot be taken by multiple supplier..so 1 to many relationship?

Reply:
Thanks Martha

Reply:
The way you've implemented it the supplier can only take one order.
If you want one order to be taken by one supplier but one supplier can take many orders, you need to put the supplier_id in the orders table instead of the order_id in the supplier table.
Does it make sense?
Try to do the excel thing that i did in the classes, where you create some data and see how it goes :)

Reply:
sure marta, will do, Thanks